src/java/org/apache/cassandra/locator/InetAddressAndPort.java

```diff
@@ -347,17 +345,10 @@ public void serialize(InetSocketAddress endpoint, DataOutputPlus out, int versio
 
         void serialize(byte[] address, int port, DataOutputPlus out, int version) throws IOException
         {
-            if (version >= MessagingService.VERSION_40)
-            {
-                out.writeByte(address.length + 2);
-                out.write(address);
-                out.writeShort(port);
-            }
-            else
-            {
-                out.writeByte(address.length);
-                out.write(address);
-            }
+            assert version >= MessagingService.VERSION_40;
```

Comment:
Totally optional since it's unrelated to the changes, but we can remove the unneeded `@SuppressWarnings("UnstableApiUsage")` class-level annotation.